include/kelcoro/thread_pool.hpp

```diff
@@ -81,8 +106,45 @@ struct task_queue {
     assert(!!nodes);
     return nodes;
   }
+
+  // precondition: node && node->task && node.status == ok
+  // executor interface
+  void attach(task_node* node) noexcept {
+    assert(node && node->task && node->status == schedule_errc::ok);
+    push(node);
+  }
 };
 
+static_assert(executor<task_queue>);
+
+namespace noexport {
+
+// We push the deadpill, and behind it ourselves,
+// in a chain, so that the deadpill is definitely alive.
+struct deadpill_pusher {
+  task_queue& queue;
+  task_node this_node;
+  task_node pill = task_node::deadpill();
+
+  static bool await_ready() noexcept {
+    return false;
+  }
+  void await_suspend(std::coroutine_handle<> handle) {
+    this_node.task = handle;
```

Comment:
need a comment to clarify why there are additional logic with creating a list. (avoiding data race)